cli/src/commands/git.rs

```diff
@@ -790,11 +791,13 @@ fn do_git_clone(
     Ok((workspace_command, stats))
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-enum BranchMoveDirection {
-    Forward,
-    Backward,
-    Sideways,
+#[derive(Clone, Debug, Eq, PartialEq)]
+enum BranchPushLabel {
+    MoveForward { old: String, new: String },
+    MoveBackward { old: String, new: String },
+    MoveSideways { old: String, new: String },
+    Add(String),
+    Delete { old: String },
```

Comment:
I don't think mixing UI logic like this is useful.

Let's say we had jj-lib function that classifies branch movements (as well as conflicts), it might return
```rust
enum BranchPushUpdate {
    Move { old: CommitId, new: CommitId, movement: Forward|Backward|Sideways },
    Add(CommitId),
    Delete(CommitId),
}
```
I don't know if this data structure is useful, but it might make sense if we have more than one `jj <forge> push` commands (or backend needs to know the movement types.) OTOH, `BranchPushLabel` is basically a UI string constructed from `BranchPushUpdate`.

Reply:
I'll just close this then.

I was wondering whether to use String vs CommitId, but I thought them essentially equivalent; we could always change one to the other if it started to matter. I thought it might be a mild improvement to collect most of the UI logic in fewer places, but it doesn't matter much, and perhaps you have your own ideas of what you want to do with this.